### PR TITLE
[openapi-gen] Add struct support to OpenAPI

### PR DESCRIPTION
**Pull request by jaylees14**
Right then... add struct support to OpenAPI.

This reuses as much of the old code as possible, but making it work for structs and services using inheritance.

It also fixes two issue: 
- the ID field in the URL was a number, not `uuid`
- the list endpoint didn't have a 401 error.

I've also noticed that 401 errors are added even when a project doesn't use auth, so I'll rectify that in a follow up PR.

### CODE REVIEW

**Review by c-harding — APPROVED** [review]
lgtm

File: src/main/scala/temple/generate/target/openapi/OpenAPIGenerator.scala [hunk 1]
```diff
@@ -37,18 +38,28 @@ private class OpenAPIGenerator private (
...
-      description = s"ID of the $lowerName to perform operations on",
-    )
-    paths.getOrElseUpdate(url, Path.Mutable(parameters = Seq(parameter))).handlers
+    val parentIDParameter = when(url.contains("{parent_id}")) {
+      Parameter(
+        InPath,
+        name = "parent_id",
+        required = Some(true),
+        schema = OpenAPISimpleType("string", "uuid"),
+        description = s"ID of the parent entity to perform operations on",
```
Comment (c-harding) [inline src/main/scala/temple/generate/target/openapi/OpenAPIGenerator.scala]: not sure about this description

**Review by c-harding — APPROVED** [review]
looks good, just a description that seems a little odd

File: src/main/scala/temple/generate/target/openapi/OpenAPIGenerator.scala (hunk 1 above)
Comment (jaylees14) [reply src/main/scala/temple/generate/target/openapi/OpenAPIGenerator.scala]: What would you suggest?

Comment (Lewiky) [reply src/main/scala/temple/generate/target/openapi/OpenAPIGenerator.scala]: "ID of the parent entity of the entity to perform operations on"?

**Review by Lewiky — APPROVED** [review]
Surprisingly small diff! nice one

**Pull request merged**